libr/bin/p/bin_ne.c

```diff
@@ -0,0 +1,133 @@
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	r_return_val_if_fail (bf && bin_obj && buf, false);
+	r_bin_ne_obj_t *res = r_bin_ne_new_buf (buf, bf->rbin->verbose);
+	if (res) {
+		*bin_obj = res;
+		return true;
+	}
+	return false;
+}
+
+static void destroy (RBinFile *bf) {
+	r_bin_ne_free (bf->o->bin_obj);
+}
+
+static void header(RBinFile *bf) {
+	struct r_bin_t *rbin = bf->rbin;
+	r_bin_ne_obj_t *ne = bf->o->bin_obj;
+	rbin->cb_printf ("Signature: NE\n");
+	rbin->cb_printf ("MajLinkerVersion: %d\n", ne->ne_header->MajLinkerVersion);
+	rbin->cb_printf ("MinLinkerVersion: %d\n", ne->ne_header->MinLinkerVersion);
+	rbin->cb_printf ("EntryTableOffset: 0x%04x\n", ne->ne_header->EntryTableOffset);
+	rbin->cb_printf ("EntryTableLength: %d\n", ne->ne_header->EntryTableLength);
+	rbin->cb_printf ("FileLoadCRC: %x\n", ne->ne_header->FileLoadCRC);
+	rbin->cb_printf ("ProgFlags: %d\n", ne->ne_header->ProgFlags);
+	rbin->cb_printf ("ApplFlags: %d\n", ne->ne_header->ApplFlags);
+	rbin->cb_printf ("AutoDataSegIndex: %d\n", ne->ne_header->AutoDataSegIndex);
+	rbin->cb_printf ("InitHeapSize: %d\n", ne->ne_header->InitHeapSize);
+	rbin->cb_printf ("InitStackSize: %d\n", ne->ne_header->InitStackSize);
+	rbin->cb_printf ("EntryPointCSIndex: %d\n", ne->ne_header->csEntryPoint);
+	rbin->cb_printf ("EntryPointIPOff: 0x%04x\n", ne->ne_header->ipEntryPoint);
+	rbin->cb_printf ("InitStack: %d\n", ne->ne_header->InitStack);
+	rbin->cb_printf ("SegCount: %d\n", ne->ne_header->SegCount);
+	rbin->cb_printf ("ModuleRefsCount: %d\n", ne->ne_header->ModRefs);
+	rbin->cb_printf ("NonResNamesTblSiz: 0x%x\n", ne->ne_header->NoResNamesTabSiz);
+	rbin->cb_printf ("SegTableOffset: 0x%x\n", ne->ne_header->SegTableOffset);
+	rbin->cb_printf ("ResourceTblOff: 0x%x\n", ne->ne_header->ResTableOffset);
+	rbin->cb_printf ("ResidentNameTblOff: 0x%x\n", ne->ne_header->ResidNamTable);
+	rbin->cb_printf ("ModuleRefTblOff: 0x%x\n", ne->ne_header->ModRefTable);
+	rbin->cb_printf ("ImportNameTblOff: 0x%x\n", ne->ne_header->ImportNameTable);
+	rbin->cb_printf ("OffStartNonResTab: %d\n", ne->ne_header->OffStartNonResTab);
+	rbin->cb_printf ("MovEntryCount: %d\n", ne->ne_header->MovEntryCount);
+	rbin->cb_printf ("FileAlnSzShftCnt: %d\n", ne->ne_header->FileAlnSzShftCnt);
+	rbin->cb_printf ("nResTabEntries: %d\n", ne->ne_header->nResTabEntries);
+	rbin->cb_printf ("OS: %s\n", ne->os);
+	rbin->cb_printf ("OS2EXEFlags: %x\n", ne->ne_header->OS2EXEFlags);
+	rbin->cb_printf ("retThunkOffset: %d\n", ne->ne_header->retThunkOffset);
+	rbin->cb_printf ("segRefThunksOff: %d\n", ne->ne_header->segrefthunksoff);
+	rbin->cb_printf ("mincodeswap: %d\n", ne->ne_header->mincodeswap);
+	rbin->cb_printf ("winver: %d.%d\n", ne->ne_header->expctwinver[1], ne->ne_header->expctwinver[0]);
+}
+
+RBinInfo *info(RBinFile *bf) {
+	r_bin_ne_obj_t *ne = bf->o->bin_obj;
+	RBinInfo *i = R_NEW0 (RBinInfo);
+	if (!i) {
```

Comment:
i prefer to use positive conditionals and save 1 line by removing this return null